Add Krator assessment (#58)
cncf/techdocs: assessments/0003-krator.md

## Introduction

This is an assessment of your project documentation and website (if present), prepared for you and your project by the CNCF techdocs team.

This document aims to:

- Measure existing documentation quality against the CNCF’s standards
- Recommend specific and general improvements
- Provide examples of great documentation as reference 
- Identify key areas which will net the largest improvement if addressed 

Since the Krator documentation site is just getting started, we'll have to create or significantly develop the current content. This assessment provides a starting point for articulating current state and ideas for the future.

## How this document works

The assessment is divided into three sections: 

- **Project documentation:** for end users of the project; aimed at people who intend to use it
- **Contributor documentation:** for new and existing contributors to the project
- **Website:** branding, website structure, and maintainability

Within each section you receive a rating on different criteria. The full definition of each criteria is defined in [the criteria](criteria.md). We recommend reading each criterion’s definition.

## Project documentation 

> | Criteria | 1 = (Is not present or requires significant work) | 2 | 3 = (is present, but needs work) | 4 | 5 = (is executed extremely well or no improvement required) |
> | ---                        | --- | --- | --- | --- | --- |
> | Information architecture   |  ✅  |     |     |    |     |
> | New user content           |  ✅  |     |     |    |     |
> | Content maintainability    |  ✅  |     |     |    |     |
> | Content creation processes |  ✅  |     |     |    |     |

Reviewer: Because the titles in this table are so unwieldy, in the Notary assessment, I've set them up like this:
(I'm not sure this is better, but it's a little easier to work with)

```suggestion
| Criteria                   | 1   | 2   | 3   | 4   | 5   |
| ---                        | --- | --- | --- | --- | --- |
| Information architecture   | ✅  |     |     |     |     |
| New user content           | ✅  |     |     |     |     |
| Content maintainability    | ✅  |     |     |     |     |
| Content creation processes | ✅  |     |     |     |     |

 Criteria: 
 - 1 = (Is not present or requires significant work) 
 - 3 = (is present, but needs work) 
 - 5 = (is executed extremely well or no improvement required)
```

        - [A Fistful of States: More State Machine Patterns in Rust](https://deislabs.io/posts/a-fistful-of-states/)
        - [Crate krator API Documentation](https://docs.rs/krator/0.4.0/krator/). This is a substantial part of the current documentation and could be incorporated into the main body (site/repo) of the documentation. This is already a great resource for users, but if we transport it to a documentation repo, it will take some work if it's not already in markdown. Is there a way extract/download these guides quickly?
        - [GitHub Documentation](https://github.com/krator-rs/krator/tree/main/docs). The Community guides teach contributing and development. 
            - The [Developer Guide](https://github.com/krator-rs/krator/blob/main/docs/community/developers.md) covers prerequisites and building, which would make a good introductory or Getting Started entry point.
            - The [Release Checklist](https://github.com/krator-rs/krator/blob/main/docs/community/release-checklist.md) appears aimed at developers of Krator, as opposed to first-time users. This should ideally be separated from the entry point.

- **New user content**:

>     - We'll need to create a clear entry point for the new user. Some of this info coud be taken from the [Developer Guide](https://github.com/krator-rs/krator/blob/main/docs/community/developers.md).

Reviewer: ```suggestion
    - We'll need to create a clear entry point for the new user. Some of this info could be taken from the [Developer Guide](https://github.com/krator-rs/krator/blob/main/docs/community/developers.md).
```

         
- **Content maintainability**:
    - Since we'll be creating a site, search doesn't apply yet (though will be part of the basic site).
    - Using Hugo will add the benefit of localization support.

> <!-- borrowed from @celestehorgan's excellent work in 0001-contour.md -->

Reviewer: I may borrow this too... 🙂

## How this document works

The assessment is divided into three sections: 

- **Project documentation:** for end users of the project; aimed at people who intend to use it
- **Contributor documentation:** for new and existing contributors to the project
- **Website:** branding, website structure, and maintainability

Within each section you receive a rating on different criteria. The full definition of each criteria is defined in [the criteria](criteria.md). We recommend reading each criterion’s definition.

## Project documentation 

| Criteria                   | 1    | 2   | 3   | 4   | 5   |
| ---                        | ---  | --- | --- | --- | --- |
| Information architecture   |  ✅  |     |     |     |     |
| New user content           |  ✅  |     |     |     |     |
| Content maintainability    |  ✅  |     |     |     |     |
| Content creation processes |  ✅  |     |     |     |     |

 Criteria: 
 - 1 = (Is not present or requires significant work) 
 - 3 = (is present, but needs work) 
 - 5 = (is executed extremely well or no improvement required)

### Comments

> - **Information Architecture**: 

Reviewer 2: Nicely done!

    - Documentation is currently in several locations and will need to be brought into one repo. The current resources are:
        - The project [README](https://github.com/krator-rs/krator)
        - [Introduction to krator Blog Post](https://deislabs.io/posts/introducing-krator/)
        - [A Fistful of States: More State Machine Patterns in Rust](https://deislabs.io/posts/a-fistful-of-states/)

>         - [Crate krator API Documentation](https://docs.rs/krator/0.4.0/krator/). This is a substantial part of the current documentation and could be incorporated into the main body (site/repo) of the documentation. This is already a great resource for users, but if we transport it to a documentation repo, it will take some work if it's not already in markdown. Is there a way extract/download these guides quickly?

Reviewer 2: One thing to note about their API documentation is that it is auto-generated as a part of the Rust Crate. While this offers convenience for the developers, it makes it difficult for non-developer users to contribute to the documentation. One thing to talk about with the devs is if they're okay with this tradeoff.

        - [GitHub Documentation](https://github.com/krator-rs/krator/tree/main/docs). The Community guides teach contributing and development. 
            - The [Developer Guide](https://github.com/krator-rs/krator/blob/main/docs/community/developers.md) covers prerequisites and building, which would make a good introductory or Getting Started entry point.
            - The [Release Checklist](https://github.com/krator-rs/krator/blob/main/docs/community/release-checklist.md) appears aimed at developers of Krator, as opposed to first-time users. This should ideally be separated from the entry point.

- **New user content**:

>     - We'll need to create a clear entry point for the new user. Some of this info could be taken from the [Developer Guide](https://github.com/krator-rs/krator/blob/main/docs/community/developers.md).

Reviewer 2: Any thoughts/guidance on what content needs to be in the entry point/getting started guide? It would be great to get more specific with them.

While there is an issue backlog, there aren't any issues with a `good first issue` label. As an invitation to new contributors, consider filing issues that are easy to remedy, labeling them accordingly, posting them on social media, and being available to help contributors through the process.  

There is a brief mention of the PR review process in [CONTRIBUTING.md](https://github.com/krator-rs/krator/blob/main/CONTRIBUTING.md), however, giving more detail about the process, such as rounds of review, time frame, a link to [CODEOWNERS](https://github.com/krator-rs/krator/blob/main/CODEOWNERS), tests that have to pass, etc. help manage contributor expectations. This, along with linking to the Code of Conduct as you currently do, establishes a context for new contributors. 

## Website

| Criteria | 1 = (Is not present or requires significant work) | 2 | 3 = (is present, but needs work) | 4 | 5 = (is executed extremely well or no improvement required) |
| ---                       | --- | --- | --- | --- | --- |
| Branding and design  |  ✅  |     |     |     |     |
| Case studies/social proof |  ✅  |     |     |    |     |
| Maintenance planning |  ✅  |     |     |     |     |

> ### Comments

Reviewer 2: ```suggestion
### Comments

Note: as Krator does not have a website yet, we did not expect any of these to be done!
```

Reviewer 2: Also: I added this just to soften the blow for the Krator team 😅. It might be a bit difficult for a small team with no resources to receive "bad marks" so to speak for something they don't already have, hence this.

- **Branding and design**: Krator will need a logo if one doesn't exist yet. Along with the logo, well need a few colors, typically inspired by the logo, for backgrounds, links, and design elements such as borders.

> Sometimes Krator is capitalized, yet other times lowercase. I would recommend using one or the other consistently.

Reviewer 2: Not a review comment but: sync with Amye on getting Krator a logo (she should already be aware of this)

> - **Case studies/social proof**: A list of case studies or companies/organizations that use Krator would be an excellent addition to the home page. If adoption is still in its early phases, I'd recommend building out a place for logos that could be added in (or uncommented) as adoption rises.

Reviewer 2: Because Krator is a new project, they likely won't have these available for a _long_ time. Instead, I'd consider:

```suggestion
- **Case studies/social proof**: As you develop Krator, keep an eye out for companies and projects using Krator, and start working with them early to do things like write blog posts about their experiences. As you gather user blog posts over time, submit a ticket to the CNCF Service Desk and we can turn these into a more marketing-friendly case studies section on your site.
```

Blog posts are a little more lightweight and easier to manage for a small project. This also makes it so that we don't build out a section that might take many months/years to become useful.

<!-- borrowed from @celestehorgan's excellent work in 0001-contour.md -->
- **Maitenence planning**: 
    - **Monorepos**: Having a docs+code monorepo is risky in the long term, as it couples all docs builds with code builds and vice versa. If docs CI fails because Netlify is temporarily down, for example, this means that all your code builds can potentially fail as well. Coupling docs in a repository with code can also make a code repo's size expand exponentially, especially as projects pick up steam, write more blog posts (with images), and add other multimedia. For these reasons, we recommend housing the documentation in a separate repo.

## Final Recommendations

### Creating documentation for new users and new contributors

1. Create a landing page that distills at-a-glance how Krator makes the world a better place, how easy it is to use, and how supportive the community is.

1. Create an About page or blog post that tells the Krator back story.

> 1. One of the most important areas for a documentation site is the starting point. I'd recommend an introductory page that explains what Krator does and why it was created that a beginner developer could understand. 

Reviewer 2: ```suggestion
1. One of the most important areas for a documentation site is the getting started guide. I'd recommend an introductory page that explains what Krator does and why it was created that a beginner developer could understand. 
```
